src/main/java/com/facebook/openwifirrm/ucentral/operationelement/HTOperationElement.java

```diff
@@ -0,0 +1,206 @@
+	/** False in a primary beacon. True in an STBC beacon. */
+	private final boolean stbcBeacon;
+	/**
+	 * Indicates the HT-MCS values that are supported by all HT STAs in the BSS. A
+	 * bitmap where a bit is set to 1 to indicate support for that MCS and 0
+	 * otherwise, where bit 0 corresponds to MCS 0.
+	 */
+	private final byte[] basicHtMcsSet;
+
+	/**
+	 * Constructs an {@code HTOperationElement} using the given field values. See
+	 * 802.11 for more details.
+	 * <p>
+	 * For details about the parameters, see the javadocs for the corresponding
+	 * member variables.
+	 */
+	public HTOperationElement(byte primaryChannel, byte secondaryChannelOffset, boolean staChannelWidth,
+			boolean rifsMode, byte htProtection, boolean nongreenfieldHtStasPresent, boolean obssNonHtStasPresent,
+			byte channelCenterFrequencySegment2, boolean dualBeacon, boolean dualCtsProtection, boolean stbcBeacon) {
+		/*
+		 * XXX some combinations of these parameters may be invalid as defined by
+		 * 802.11-2020, but this is not checked here. If fidelity to 802.11 is required,
+		 * the caller of this method must make sure to pass in valid parameters. The
+		 * 802.11-2020 specification has more details about the parameters.
+		 */
+		this.primaryChannel = primaryChannel;
+		this.secondaryChannelOffset = secondaryChannelOffset;
+		this.staChannelWidth = staChannelWidth;
+		this.rifsMode = rifsMode;
+		this.htProtection = htProtection;
+		this.nongreenfieldHtStasPresent = nongreenfieldHtStasPresent;
+		this.obssNonHtStasPresent = obssNonHtStasPresent;
+		this.channelCenterFrequencySegment2 = channelCenterFrequencySegment2;
+		this.dualBeacon = dualBeacon;
+		this.dualCtsProtection = dualCtsProtection;
+		this.stbcBeacon = stbcBeacon;
+		// the next 16 bytes are for the basic HT-MCS set
+		// a default is chosen; if needed, we can add a parameter to set these
+		this.basicHtMcsSet = new byte[16];
+	}
+
+	/** Constructor with the most used parameters. */
+	public HTOperationElement(byte primaryChannel, byte secondaryChannelOffset, boolean staChannelWidth,
+			byte channelCenterFrequencySegment2) {
+		this(primaryChannel, secondaryChannelOffset, staChannelWidth, false, (byte) 0, true, false,
+				channelCenterFrequencySegment2, false, false, false);
+	}
+
+	/**
+	 * Constructs an {@code HTOperationElement} by decoding {@code htOper}.
+	 *
+	 * @param htOper a base64 encoded properly formatted HT operation element (see
+	 *               802.11)
+	 */
+	public HTOperationElement(String htOper) {
+		byte[] bytes = Base64.decodeBase64(htOper);
+		this.primaryChannel = bytes[0];
+		this.secondaryChannelOffset = (byte) (bytes[1] >>> 6);
+		this.staChannelWidth = ((bytes[1] & 0b00100000) >>> 5) == 1;
+		this.rifsMode = ((bytes[1] & 0b00010000) >>> 4) == 1;
+		this.htProtection = (byte) (bytes[2] >>> 6);
+		this.nongreenfieldHtStasPresent = ((bytes[2] & 0b00100000) >>> 5) == 1;
+		this.obssNonHtStasPresent = ((bytes[2] & 0b00001000) >>> 4) == 1;
```

Comment:
should be >>> 3

Reply:
Good catch, I will fix it in my next push.